xapian-core/include/xapian/cluster.h

```diff
@@ -0,0 +1,412 @@
+/** @file cluster.h
+ *  @brief Cluster API
+ */
+/* Copyright (C) 2010 Richard Boulton
+ * Copyright (C) 2016 Richhiey Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#ifndef XAPIAN_INCLUDED_CLUSTER_H
+#define XAPIAN_INCLUDED_CLUSTER_H
+
+#if !defined XAPIAN_IN_XAPIAN_H && !defined XAPIAN_LIB_BUILD
+#error "Never use <xapian/cluster.h> directly; include <xapian.h> instead."
+#endif
+
+#include <xapian/mset.h>
+#include <xapian/types.h>
+#include <xapian/visibility.h>
+
+#include <unordered_map>
+#include <vector>
+
+/** Maximum number of times KMeans algorithm will iterate
+ *  till it converges
+ */
+#define MAX_ITERS 1000
```

Comment:
Yes I have moved this to the kmeans.cc file and within the constructor, I check if the second parameter is 0. If it is, max_iters = MAX_ITERS or else max_iters = max_iters (parameter value passed in through constructor)